### DISCUSSION TITLE
fit for realtime ASIO reading ?

### DISCUSSION THREAD

**Discussion by mprevot** [naudio/NAudio#1154] [Q&A] [answered]
I have ASIO midi input, can I use naudio to read the ASIO midi input in realtime so I can do something with this data in c# ?

**Comment by markheath** [comment, marked as answer]
ASIO is audio only. If you mean that your sound card has a MIDI input, then you can use `WinRTMidiIn` to capture that, and then use it to produce sound that you play with ASIO
